replica/database.cc

```diff
@@ -1073,10 +1074,22 @@ db::commitlog* database::commitlog_for(const schema_ptr& schema) {
         : _commitlog.get();
 }
 
-void database::add_column_family(keyspace& ks, schema_ptr schema, column_family::config cfg, is_new_cf is_new, locator::token_metadata_ptr not_commited_new_metadata) {
+future<owned_ranges_ptr> maybe_make_owned_ranges_ptr(locator::effective_replication_map_ptr erm, locator::host_id host_id) {
+    if (!erm) {
+        co_return nullptr;
+    }
+    if (erm->get_token_metadata().sorted_tokens().empty()) {
+        co_return nullptr;
+    }
+    co_return make_owned_ranges_ptr(co_await erm->get_ranges(host_id));
```

Comment:
We could avoid this if we had erm->get_local_ranges(), which serves pre-computed data.

It would also automatically recompute on topology changes.

Also, this way we would naturally have them per-keyspace with vnodes (erm is per-keyspace) rather than per table. So less memory usage.